### PR TITLE
Feature/explicit pdfs

### PR DESCRIPTION
**Pull request by bengolder**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,7 +23,6 @@ node_modules
 staticfiles
 project/media
 local_settings.py
-*.pdf
```
Comment (bengolder) [inline .gitignore:26]: Now that we have factories for FilledPDF and FillablePDF this ensures the files needed for those factories will be included.

File: features/app_index_views.feature [hunk 2]
```diff
@@ -9,19 +9,19 @@ Feature: An org user can view applications to their organization
 
 
 	Scenario: Org user can see a list of all unread applications in the "Unread" tab
-	   Given I log in as an org user at "cc_pubdef"
+	 Given I log in as an org user at "cc_pubdef"
```
Comment (bengolder) [inline features/app_index_views.feature:12]: I don't understand how whitespace works in behave tests. I was trying to make these consistent. ¯\_(ツ)_/¯

File: features/apply.feature [hunk 3]
```diff
@@ -1,33 +1,33 @@
 Feature: User can Apply to CMR
   Scenario: User Successfully applies to Contra Costa
-    Given that "apply/" loads
+    Given that "/apply/" loads
```
Comment (bengolder) [inline features/apply.feature:3]: I realized we had inconsistent and fragile ways of checking URLs, so I tried to ensure they all included the first slash and that comparisons were more able to handle deviations.

File: features/steps/county_applications_steps.py [hunk 4]
```diff
@@ -1,18 +1,17 @@
-from behave import when
+from behave import when, given
 from intake import models
+from features.steps.language_hacks import oxford_comma_text_to_list
```
Comment (bengolder) [inline features/steps/county_applications_steps.py:3]: I made this new function so we can use it elsewhere in behave tests.

File: features/steps/pdf_steps.py [hunk 5]
```diff
@@ -24,21 +26,22 @@ def test_prefilled_pdf_for_applicant(context, applicant_name):
...
+    applicant_names = oxford_comma_text_to_list(applicant_names)
+    app_ids = []
+    for applicant_name in applicant_names:
+        first_name, last_name = applicant_name.split(' ')
+        app_ids.append(
+            models.Application.objects.filter(
+                form_submission__first_name=first_name,
+                form_submission__last_name=last_name).first().id)
+    prebuilt = pdf_service.get_prebuilt_pdf_bundle_for_app_id_set(app_ids)
+    context.test.assertTrue(prebuilt)
```
Comment (bengolder) [inline features/steps/pdf_steps.py:40]: The above step function checks for a prebuilt PDF _with a precise match_ for the given set of applicants. If one name is off from the result or the result is missing a name, this test step will fail.

File: features/steps/web_loads_and_contains_steps.py [hunk 6]
```diff
@@ -1,19 +1,45 @@
...
+    element.click()
+
+
 @then('it should load "{url}"')
 def test_page_loads(context, url):
     browser_url = urlparse(context.browser.current_url)
-    context.test.assertEquals(url, browser_url.path[1:])
+    expected_url = urlparse(urljoin(context.test.live_server_url, url))
+    context.test.assertEquals(
+        expected_url.path, browser_url.path)
```
Comment (bengolder) [inline features/steps/web_loads_and_contains_steps.py:32]: These changes ensure more robust url equality comparison

File: intake/models/form_submission.py [hunk 7]
```diff
@@ -226,50 +222,6 @@ def get_counties(self):
...
-        display_form.display_template_name = "formation/intake_display.jinja"
-        display_form.submission = self
-        show_declaration = any(self.organizations.all().values_list(
-            'requires_declaration_letter', flat=True))
-        if show_declaration:
-            declaration_letter_form = DeclarationLetterDisplay(
-                init_data, validate=True)
-            return display_form, declaration_letter_form
-        return display_form, None
-
```
Comment (bengolder) [inline intake/models/form_submission.py:272]: Removed this code from it's position as a spaghetti model method and refactored it into several functions in `display_form_service`

File: intake/services/applications_service.py [hunk 8]
```diff
@@ -122,3 +122,14 @@ def get_serialized_application_history_events(application, user):
     else:
         status_updates = get_status_updates_for_org_user(application)
     return serializers.StatusUpdateSerializer(status_updates, many=True).data
+
+
+def mark_opened(application_ids):
+    # This does not fire events
+    apps = models.Application.objects.filter(id__in=application_ids)
+    apps.update(has_been_opened=True)
```
Comment (bengolder) [inline intake/services/applications_service.py:130]: Event integration has been postponed until merge

File: intake/services/applications_service.py [hunk 9]
```diff
@@ -122,3 +122,14 @@ def get_serialized_application_history_events(application, user):
...
+
+
+def get_valid_application_ids_from_set(application_ids):
+    return models.Application.objects.filter(
+        id__in=application_ids).values_list('id', flat=True)
```
Comment (bengolder) [inline intake/services/applications_service.py:135]: This just validates a set of IDs. It's useful for query-param-based sets of IDs.

File: intake/services/events_service.py [hunk 10]
```diff
@@ -183,3 +183,26 @@ def partnership_interest_submitted(partnership_lead):
...
+        distinct_id=request.user.get_uuid(),
+        event_name=user_event_name,
+        organization_name=view.organization.name)
+    for app in view.applications:
+        log_to_mixpanel.delay(
+            distinct_id=app.get_uuid(),
+            event_name=applicant_event_name,
+            bundle_organization_name=app.organization.name,
+            user_email=request.user.email,
+            user_organization_name=request.user.profile.organization.name)
```
Comment (bengolder) [inline intake/services/events_service.py:208]: These are just sketches for potential events. I assumed rewriting or refactoring would be necessary post-merge.

Comment (jennifermarie) [reply intake/services/events_service.py:208]: Yep, 100%. Doing some local reconciliation between this & the events branch. Thank you!

File: features/app_index_views.feature (hunk 2 above)
Comment (jennifermarie) [reply features/app_index_views.feature:12]: It's the worst. I'm doing spot checks on all of these because I've had similar failures.

**Pull request merged**